### PR TITLE
fix: update `prepare_dynamic_op_kwargs` to not create overlapping wires by adding an accumulator

### PR DESCRIPTION
**Pull request by andrijapau**
**Context:**

I am seeing these kinds of errors for various workflows.
```
RuleLoweringWarning: Failed to lower the registered adjoint decomposition rules for Adjoint(QROM){bitstrings:[tensor<210x12xi64>]}{control_wires:8,target_wires:12,work_wires:7}{clean = true}: control_wires and target_wires must not overlap, got [0, 1, 2, 3, 4, 5, 6, 7] and [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
```

**Description of the Change:**

Update the logic to follow the strategy in https://github.com/PennyLaneAI/catalyst/pull/3214.

**Benefits:** Cannot generate overlapping wires

**Possible Drawbacks:** None identified.

[sc-130713]

### CODE REVIEW

File: frontend/catalyst/decomposition/decomposition_rules.py [hunk 1]
```diff
@@ -378,8 +379,13 @@ def prepare_dynamic_op_kwargs(dynamic_shape, wire_lens) -> dict:
         dict: argument names to dummy values
     """
     kwargs = {}
+
+    # NOTE: Run an accumulator to generate unique negative wire labels.
+    # Otherwise, operators with control and target wires as their arguments
+    # will receive overlapping wires.
+    wire_counter = itertools.count(-1, -1)
```
Comment (andrijapau) [inline frontend/catalyst/decomposition/decomposition_rules.py:386]: Does it matter if these are negative or positive?

File: frontend/test/pytest/test_decomposition.py [hunk 2]
```diff
@@ -263,7 +281,11 @@ def test_wrapper_passes_compilable_data_to_conditions(self, mocker):
         assert call_kwargs["a"] is True
         assert call_kwargs["b"] == 3.14
         assert call_kwargs["thing"] == "string"
-        assert call_kwargs["wires"].tolist() == [0, 1]
+
+        probe_wires = np.asarray(call_kwargs["wires"])
```
Comment (andrijapau) [inline frontend/test/pytest/test_decomposition.py:285]: I think this is a more robust test as we don't care about the specific wire labels, just that they are unique and negative

**Review by kipawaa — APPROVED** [review]
Nice!

**Review by maliasadi — APPROVED** [review]
🐛
